src/components/Input.react.js

```diff
@@ -25,25 +25,29 @@ export default class Input extends Component {
             setProps,
             type
         } = this.props;
-        const {value} = this.state;
+        //const {value} = this.state;
         return (
             <input
-                onChange={e => {
-                    this.setState({value: e.target.value});
-                    if (setProps) {
-                        if (type === 'number') {
-                            setProps({value: Number(e.target.value)});
-                        }
-                        else {
-                            setProps({value: e.target.value});
+                onBlur={
+                    event => {
+                        this.setState({value: event.target.value});
+                        if (setProps) {
+                            if (type === 'number') {
+                                setProps({value: Number(event.target.value)});
+                            }
+                            else {
+                                setProps({value: event.target.value});
+                            }
                         }
+                        if (fireEvent) fireEvent({event: 'blur'});
+                    }
+                }
+                onChange={
+                    () => {
+                        if (fireEvent) fireEvent({event: 'change'});
                     }
-                    if (fireEvent) fireEvent({event: 'change'});
-                }}
-                onBlur={() => {
-                    if (fireEvent) fireEvent({event: 'blur'});
-                }}
-                value={value}
+                }
+                //value={value}
```

Comment:
The reason that this doesn't work without removing the `value={value}`, is probably because `onBlur` only fires the update, well, on blur, as opposed to `onChange` which fires every change. Setting `value` here will overwrite anything you put in the input (try having just `<input value={value} />` in there, and you'll see what I mean - it is as if it's 'locked' to whatever is put in as the `value` prop) so `onBlur`will fire it's update on blur, but because the input is 'locked', it fires the update with the same value. Does that make sense?

Reply:
Yes it make sense, maybe it could work with another variable, i.e. `typed` for example. I will try